PyInstaller/utils/hooks/gi.py

```diff
@@ -306,3 +310,111 @@ def collect_glib_translations(prog, lang_list=None):
+        if dest_name.parent != SCHEMA_DEST_DIR:
+            output_toc.append(toc_entry)
+            continue
+
+        # It seems schemas directory contains different files with different suffices:
+        #  - .gschema.xml
+        #  - .schema.override
+        #  - .enums.xml
+        # To avoid omitting anything, simply collect everything into temporary directory.
+        # Exemptions are gschema.dtd (which should be unnecessary) and gschemas.compiled (which we will generate
+        # ourselves in this function).
+        if src_name.name in {"gschema.dtd", "gschemas.compiled"}:
+            continue
+
+        schema_files.append(src_name)
+
+    # If there are no schema files available, simply return the input datas TOC.
+    if not schema_files:
+        return datas_toc
+
+    # Ensure that `glib-compile-schemas` executable is in PATH, just in case...
+    schema_compiler_exe = shutil.which('glib-compile-schemas')
+    if not schema_compiler_exe:
+        logger.warning("GLib schema compiler (glib-compile-schemas) not found! Skipping GLib schema recompilation...")
+        return datas_toc
+
+    # If `gschemas.compiled` file already exists in the temporary working directory, record its modification time and
+    # hash. This will allow us to restore the modification time on the newly-compiled copy, if the latter turns out
+    # to be identical to the existing old one. Just in case, if the file becomes subject to timestamp-based caching
+    # mechanism.
+    compiled_file = workdir / "gschemas.compiled"
+    old_compiled_file_hash = None
+    old_compiled_file_stat = None
+
+    if compiled_file.is_file():
+        # Record creation/modification time
+        old_compiled_file_stat = compiled_file.stat()
+        # Compute SHA1 hash; since compiled schema files are relatively small, do it in single step.
+        old_compiled_file_hash = hashlib.sha1(compiled_file.read_bytes()).digest()
+
+    # Ensure that temporary working directory exists, and is empty.
+    if workdir.exists():
+        shutil.rmtree(workdir)
+    workdir.mkdir(exist_ok=True)
+
+    # Copy schema (source) files to temporary working directory
+    for schema_file in schema_files:
+        shutil.copy(schema_file, workdir)
+
+    # Compile. The glib-compile-schema might produce warnings on its own (e.g., schemas using deprecated paths, or
+    # overrides for non-existent keys). Since these are non-actionable, capture and display them only as a DEBUG
+    # message, or as a WARNING one if the command fails.
+    logger.info("Compiling collected GLib schema files in %r...", str(workdir))
+    try:
+        cmd_args = [schema_compiler_exe, str(workdir), '--targetdir', str(workdir)]
+        p = subprocess.run(
+            cmd_args,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=True,
+            universal_newlines=True,
```

Comment:
Python 3.6 is toast now so we can finally stop calling text mode `universal_newlines`.

```suggestion
            text=True,
```